src/components/map/Layers/Countries.tsx

```diff
@@ -54,80 +89,107 @@ function FilterCountryEstimates(map: mapboxgl.Map, estimateGradePrevalences: Est
     }
 }
 
-const Countries = (map: mapboxgl.Map | undefined, estimateGradePrevalences: EstimateGradePrevalence[]) => {
+const Countries = (map: mapboxgl.Map | undefined, {estimateGradePrevalences, countryFocus}: CountriesMapConfig) => {
 
     const [state] = useContext(AppContext);
     const [popup, setPopup] = useState<mapboxgl.Popup | undefined>(undefined);
-    const [currentFeatureId, setCurrentFeatureId] = useState<string | undefined>(undefined);
+    const [highlight, setHighlight] = useState<string | undefined>(undefined);
+    const history = useHistory();
 
     // If estimates are updated, waits until map is loaded then maps estimate data to country features
     useEffect(() => {
-        if (estimateGradePrevalences.length > 0 && map && map.getSource('Countries')) {
-            SetCountryEstimates(map, estimateGradePrevalences);
-        }
-        else if (map) {
-            map.on('sourcedata', function (e: any) {
-                if (e.sourceId === 'Countries' && map.isSourceLoaded('Countries')) {
+
+        if (map)
+        {
+            if(estimateGradePrevalences)
+            {
+                if (map?.getSource(COUNTRY_LAYER_ID) === undefined) 
+                {
+                    map.once('sourcedata', function (e: any) {
```

Comment:
So it looks like this logic here tries to wait for 'sourcedata' to return a sourceId twice. This seems a bit brittle to me (what if we don't find e.sourceId on the third query as well?). 

Would it be advisable for us to refactor this logic to somehow keep querying for sourcedata until we get e.sourceId?

Reply:
Nested ifs should be combined if possible

Reply:
yea there is a problem with the map loading to fast once its been cached. I need to look into refactoring this part without breaking it